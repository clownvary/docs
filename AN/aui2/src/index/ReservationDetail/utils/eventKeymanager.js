export function wrapEventIndex(eventIndex) {
  return `event_${eventIndex}`;
}

export function extractEventIndexWrapper(eventIndexWrapper) {
  return eventIndexWrapper.replace('event_', '');
}
