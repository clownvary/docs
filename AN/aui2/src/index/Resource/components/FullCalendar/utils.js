
import moment from 'moment';
import DateTimeFormat from 'shared/utils/DateTimeFormat';

const {
  permitStartEventDate,
  dateFormat,
  currentDate,
  currentTime,
  permit_label: permitLabel,
  activity_label: activityLabel,
  daycare_label: daycareLabel,
  company_label: companyLabel
} = window.__resourceCalender__.__initialState__;

export const START_EVENT_DATE = permitStartEventDate;
export const DATE_FORMAT = dateFormat;

export const CURRENT_DATE = currentDate;
export const CURRENT_TIME = currentTime;

export const PERMIT_LABEL = permitLabel;
export const ACTIVITY_LABEL = activityLabel;
export const DAYCARE_LABEL = daycareLabel;
export const COMPANY_LABEL = companyLabel;

export const CURRENT_SERVER_DATE_TIME = DateTimeFormat.parseDateTime(`${CURRENT_DATE} ${CURRENT_TIME}`);

export const bookingHelper = {
  list: new Map(),
  /**
   * @type {Jquery Object}
   */
  alllistDoms: null,
  /**
   * @type {boolean}
   * @desc Determines the selection booking action.
   * - true= SINGLE
   * - false= MULTIPLE
   */
  selectSingle: true,

  addItem(key, domValue) {
    this.list.set(key, domValue);
    domValue.addClass('is-selected');
    return this;
  },

  deleteItem(key) {
    this.list.get(key).removeClass('is-selected');
    this.list.delete(key);
    return this;
  },

  clear() {
    this.list.clear();
    if (this.alllistDoms) {
      this.alllistDoms.find('.is-selected').removeClass('is-selected');
    }
    return this;
  },

  setRootDom($el) {
    this.alllistDoms = $el;
    return this;
  },

  get events() {
    return [...this.list.keys()];
  },
 // Single selection action;
  select(key, domValue) {
    if (this.selectSingle) {
      this.clear();
      this.addItem(key, domValue);
      return this;
    }
    if (this.list.has(key)) {
      this.deleteItem(key);
    } else {
      this.addItem(key, domValue);
    }
    return this;
  }
};

export const createCalendarTimer = (startTime = CURRENT_SERVER_DATE_TIME) => {
  const latestTime = moment(startTime);
  let timer = null;

  const start = () => {
    timer = setInterval(() => {
      latestTime.add(1, 's');
    }, 1000);
  };

  const stop = () => {
    if (timer) {
      clearInterval(timer);
    }
  };

  const getTime = () => latestTime;

  return {
    start,
    stop,
    getTime
  };
};
