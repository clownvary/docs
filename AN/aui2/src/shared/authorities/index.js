
import Authority from './Authority';
import AuthorityID from './AuthorityID';
import AuthorityType from './AuthorityType';
import AuthorityUIType from './AuthorityUIType';

export default {
  Authority,
  AuthorityID,
  AuthorityType,
  AuthorityUIType
};
