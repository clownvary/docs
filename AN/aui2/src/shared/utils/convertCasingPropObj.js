function _formatCasingKey(key) {
  let casingKey = '';
  key.split('_').forEach((s, idx) => {
    if (idx === 0) {
      casingKey = s;
    } else if (s === 'id') {
      casingKey += s.toUpperCase();
    } else {
      casingKey += s.charAt(0).toUpperCase() + s.slice(1);
    }
  });
  return casingKey;
}

export default function convertCasingPropObj(object) {
  if (Object.prototype.toString.call(object) === '[object Object]') {
    const obj = {};
    Object.keys(object).forEach((key) => {
      const value = convertCasingPropObj(object[key]);
      obj[_formatCasingKey(key)] = value;
    });
    return obj;
  } else if (Object.prototype.toString.call(object) === '[object Array]') {
    const arr = [];
    Object.keys(object).forEach((key) => {
      const value = convertCasingPropObj(object[key]);
      arr[_formatCasingKey(key)] = value;
    });
    return arr;
  }
  return object;
}

