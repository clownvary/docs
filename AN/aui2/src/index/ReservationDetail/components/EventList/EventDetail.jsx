import PropTypes from 'prop-types';
import React from 'react';
import { fromJS } from 'immutable';
import * as da from 'react-base-ui/lib/utils/dataAccess';
import UIComponent from 'shared/components/UIComponent';
import Waiver from 'shared/components/Waiver';
import Notes from 'shared/components/Notes';
import FeeSection from 'shared/components/PermitFee/FeeSection';
import feesOrderByRecurring from 'shared/utils/feesOrderByRecurring';
import { Authority } from 'shared/authorities';
import EventSurvey from './EventSurvey';
import { wrapEventIndex } from '../../utils/eventKeymanager';
import './index.less';

export default class EventDetail extends UIComponent {

  static propTypes = {
    eventID: PropTypes.string.isRequired,
    eventIndex: PropTypes.string.isRequired,
    changeQuestion: PropTypes.func.isRequired,
    deleteQuestion: PropTypes.func.isRequired,
    changeWaiverByEventID: PropTypes.func.isRequired,
    saveWaiver: PropTypes.func.isRequired,
    saveNotes: PropTypes.func.isRequired,
    permitDetailsChanged: PropTypes.func.isRequired
  };

  loadAddableWaivers = () => {
    const { batchID, receiptID, eventID, eventIndex } = this.props;
    this.props.loadAddableWaivers({
      batchID,
      receiptID,
      permitEventId: eventID,
      eventIndex
    });
  }

  addQuestion = () => {
    const { permitID, batchID, receiptID, eventID, eventIndex, newEntryID } = this.props;
    this.props.addQuestion(
      eventIndex,
      {
        permitID,
        batchID,
        receiptID,
        permitEventId: eventID,
        newEntryID
      }
    );
  }

  render() {
    const { eventDetail: detail,
      batchID,
      receiptID,
      eventID,
      newEntryID,
      permitID,
      eventIndex,
      waiverList,
      noteList,
      facility,
      allEventConfig,
      eventActionInformation: eai,
      errorHighlight,
      changeWaiverByEventID,
      saveWaiver,
      saveNotes,
      setEventValidStatus,
      setWaiverErrorMessage,
      showUpdated,
      permitDetailsChanged,
      removeWaiverConfirmChangeError,
      questions,
      hasRequiredQuestion,
      deleteQuestion,
      changeQuestion,
      initialData,
      addableQuestionsLoaded,
      addQuestionList
    } = this.props;
    const { permitLabel: permitWording } = initialData;
    const isShow = detail.get('isShow');
    const generatedID = wrapEventIndex(eventIndex);
    const currentWaiver = waiverList.get('allWaivers').get(generatedID) || {};
    const currentNotes = fromJS(noteList.get('allNotes').get(generatedID) || {});
    const currentFacilityFees = facility.get('allFacilities').get(generatedID) || [];

    const facilityFees = currentFacilityFees.size ?
      currentFacilityFees.get('eventFee').toJS().facilityFees || [] : [];
    const finalFacilityFees = feesOrderByRecurring(facilityFees);

    const allowResetFees = currentFacilityFees.getIn && currentFacilityFees.getIn(['eventSummary', 'allowResetFees']);
    const disabledEditQuestion = Authority.isDisabled(eai.editQuestion.systemBehaviorID) ||
      Authority.isHidden(eai.editQuestion.systemBehaviorID);
    const disabledDeleteQuestion = Authority.isDisabled(eai.deleteQuestion.systemBehaviorID) ||
      Authority.isHidden(eai.deleteQuestion.systemBehaviorID);
    const disabledEditWaiver = Authority.isDisabled(eai.editWaiver.systemBehaviorID) ||
      Authority.isHidden(eai.editWaiver.systemBehaviorID);
    const disabledEditNote = Authority.isDisabled(eai.editNote.systemBehaviorID) ||
      Authority.isHidden(eai.editNote.systemBehaviorID);
    const disabledAddQuestions = Authority.isDisabled(eai.addQuestions.systemBehaviorID) ||
      Authority.isHidden(eai.addQuestions.systemBehaviorID);
    const disabledAddWaivers = Authority.isDisabled(eai.addWaivers.systemBehaviorID) ||
      Authority.isHidden(eai.addWaivers.systemBehaviorID);
    const disabledResetFee = Authority.isDisabled(eai.resetFees.systemBehaviorID) ||
    Authority.isHidden(eai.resetFees.systemBehaviorID);

    const hideAddCharge = !Authority.isHidden(eai.addCharge.systemBehaviorID);
    const hideDeleteCharge = !Authority.isHidden(eai.deleteCharge.systemBehaviorID);
    const hideEditCharge = !Authority.isHidden(eai.editCharge.systemBehaviorID);

    return (
      <div className={`event-detail ${isShow.get(generatedID) ? 'reservation-fee' : 'u-hidden'} ${errorHighlight ? 'content-error' : ''}`}>
        {
          allEventConfig.get('hideReservationCharges') ?
          '' :
          <div className="permit-fee">
            <div className="reservation-fee">
              <FeeSection
                batchID={batchID}
                receiptID={receiptID}
                newEntryID={newEntryID}
                eventID={eventID}
                eventIndex={eventIndex}
                facilityFees={finalFacilityFees}
                fetchPermitFee={params => this.props.fetchReservationFeeThenUpdate(params)}
                permitDetailsChanged={permitDetailsChanged}
                feeActionStatus={{
                  allowAddFee: hideAddCharge,
                  allowDeleteFee: hideDeleteCharge,
                  allowEditFee: hideEditCharge
                }}
                allowResetFees={allowResetFees}
                disabledResetFee={disabledResetFee}
                deleteReservationFeeDetail={this.props.deleteReservationFeeDetail}
                resetFeeAsyncAction={this.props.resetFeeAsyncAction}
                pagination={this.props.pagination}
                showMore={this.props.showMore}
                applyToAll={this.props.applyToAll}
                detectSameCharge={this.props.detectSameCharge}
              />
            </div>
          </div>
        }
        <div className="event-detail__question leftMargin">
          <EventSurvey
            eventID={eventID}
            eventIndex={eventIndex}
            newEntryID={newEntryID}
            permitID={permitID}
            showUpdated={showUpdated}
            questions={questions}
            hideCustomQuestionsSection={allEventConfig.get('hideCustomQuestionsSection')}
            hasRequiredQuestion={hasRequiredQuestion}
            readOnly={disabledEditQuestion}
            canDelete={!disabledDeleteQuestion}
            changeQuestion={changeQuestion}
            deleteQuestion={deleteQuestion}
            addQuestion={this.addQuestion}
            addableQuestionsLoaded={addableQuestionsLoaded}
            addQuestionList={addQuestionList}
            disabledAddQuestions={disabledAddQuestions}
          />
        </div>
        {
          allEventConfig.get('hideChecklistItemsSection') ?
          '' :
          <div className="event-detail__waiver leftMargin">
            <Waiver
              permitWording={permitWording}
              waiver={waiverList}
              data={da.get(currentWaiver, 'data')}
              hasNew={da.get(currentWaiver, 'hasNew')}
              addableWaivers={da.get(currentWaiver, 'addableWaivers')}
              addableWaiversLoaded={da.get(currentWaiver, 'addableWaiversLoaded')}
              eventID={eventID}
              eventIndex={eventIndex}
              eventDetail={detail}
              readOnly={disabledEditWaiver}
              changeWaiverByEventID={changeWaiverByEventID}
              saveWaiver={saveWaiver}
              permitDetailsChanged={permitDetailsChanged}
              setEventValidStatus={setEventValidStatus}
              setWaiverErrorMessage={setWaiverErrorMessage}
              newEntryID={newEntryID}
              showUpdated={showUpdated}
              afterAgreeWaiver={removeWaiverConfirmChangeError}
              loadAddableWaivers={this.loadAddableWaivers}
              disabledAddWaivers={disabledAddWaivers}
            />
          </div>
        }
        {
          allEventConfig.get('hideNotesSection') ?
            '' :
            <div className="event-detail__notes leftMargin">
              <Notes
                {...{
                  eventID,
                  eventIndex,
                  newEntryID,
                  saveNotes,
                  showUpdated,
                  permitDetailsChanged
                }}
                immediate
                notes={currentNotes}
                readOnly={disabledEditNote}
              />
            </div>
        }
      </div>
    );
  }
}
