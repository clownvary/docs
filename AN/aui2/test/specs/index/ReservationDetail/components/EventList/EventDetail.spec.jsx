import React from 'react';
import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';
import { fromJS } from 'immutable';
import convertCasingPropObj from 'shared/utils/convertCasingPropObj';
import EventDetail from 'index/ReservationDetail/components/EventList/EventDetail';
import { wrapEventIndex } from 'index/ReservationDetail/utils/eventKeymanager';
import FeeSection from 'shared/components/PermitFee/FeeSection';
import { Authority } from 'shared/authorities';

jest.mock('shared/components/PermitFee/FeeSection', () => 'FeeSection');

// EventDetail props begin
const createProps = (eventID = 1, overrides) => {
  const eventIndex = `${eventID}`;

  const eventDetail = {
    isShow: {
      [wrapEventIndex(eventIndex)]: true
    }
  };

  const waiverList = {
    allWaivers: {
      [wrapEventIndex(eventIndex)]: {}
    },
    data: []
  };

  const noteList = {
    allNotes: {
      [wrapEventIndex(eventIndex)]: {}
    }
  };

  const facility = {
    allFacilities: {
      [wrapEventIndex(eventIndex)]: {
        eventFee: {
          facilityFees: []
        }
      }
    }
  };

  const eventActionInformation = convertCasingPropObj(__reservationDetail__.__initialState__.reservationDetail.event_action_information);

  const allEventConfig = {
    hideReservationCharges: false,
    hideNotesSection: false,
    hideChecklistItemsSection: false,
    hideCustomQuestionsSection: false
  };

  const authorities = convertCasingPropObj(window.__reservationDetail__.__initialState__.authorities);

  const errorHighlight = true;
  const isTrrigerCustomerSearch = false;
  const isPermitUpdatingMode = false;
  const rules = [];
  const initialData = {
    permitLabel: ''
  };
  return {
    initialData,
    eventDetail: fromJS(eventDetail),
    questions: [],
    errors: [],
    hasRequiredQuestion: true,
    waiverList: fromJS(waiverList),
    noteList: fromJS(noteList),
    facility: fromJS(facility),
    allEventConfig: fromJS(allEventConfig),
    rules,
    eventIndex,
    authorities,
    errorHighlight,
    isPermitUpdatingMode,
    eventActionInformation,
    isTrrigerCustomerSearch,
    addQuestionList: [],
    ...overrides
  };
};

describe('index -> ReservationDetail -> components -> EventDetail', () => {
  Authority.init(window.__authoritiy__);
  const setup = (eventID, overrides) => mount(<EventDetail {...createProps(eventID, overrides)} />);

  it('render component correctly', () => {
    const component = mount(<EventDetail {...createProps()} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('render component correctly if no data', () => {
    const props = createProps(
      1, { eventIndex: '2', errorHighlight: false }
    );
    props.allEventConfig = props.allEventConfig.mapEntries(([k, v]) => [k, !v]);
    const component = mount(<EventDetail {...props} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('component should work fine', () => {
    const fetchReservationFeeThenUpdate = jest.fn();
    const loadAddableWaivers = jest.fn();
    const addQuestion = jest.fn();
    const component = setup(1, { fetchReservationFeeThenUpdate, loadAddableWaivers, addQuestion });
    component.instance().loadAddableWaivers();
    expect(loadAddableWaivers).toHaveBeenCalled();
    component.instance().addQuestion();
    expect(addQuestion).toHaveBeenCalled();

    component.find(FeeSection).props().fetchPermitFee();
    expect(fetchReservationFeeThenUpdate).toHaveBeenCalled();
  });
});
