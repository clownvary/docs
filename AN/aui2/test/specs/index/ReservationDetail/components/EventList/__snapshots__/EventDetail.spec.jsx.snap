// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index -> ReservationDetail -> components -> EventDetail render component correctly 1`] = `
<EventDetail
  addQuestionList={Array []}
  allEventConfig={
    Immutable.Map {
      hideReservationCharges: false,
      hideNotesSection: false,
      hideChecklistItemsSection: false,
      hideCustomQuestionsSection: false,
    }
  }
  authorities={
    Array [
      Object {
        "authorityType": "display",
        "id": "newReservationHome",
        "name": "New Reservation Home",
      },
      Object {
        "authorityType": "display",
        "id": "bookingGrid",
        "name": "Booking Grid",
      },
      Object {
        "authorityType": "enabled",
        "id": "permitActions",
        "name": "Permit Actions",
      },
      Object {
        "authorityType": "display",
        "id": "resourceCalendar",
        "name": "Resource Calendar",
      },
      Object {
        "authorityType": "enabled",
        "id": "bookingGridIcon",
        "name": "Booking Grid Icon",
      },
      Object {
        "authorityType": "enabled",
        "id": "reserveActions",
        "name": "Reserve Actions",
      },
      Object {
        "authorityType": "enabled",
        "id": "changePermitStatus",
        "name": "Change Permit Status",
      },
      Object {
        "authorityType": "disabled",
        "id": "cancelPermit",
        "name": "Cancel Permit",
      },
      Object {
        "authorityType": "disabled",
        "id": "modifyPaymentPlan",
        "name": "Cancel Permit",
      },
      Object {
        "authorityType": "enabled",
        "id": "makePayment",
        "name": "Cancel Permit",
      },
      Object {
        "authorityType": "disabled",
        "id": "refundCharges",
        "name": "Cancel Permit",
      },
      Object {
        "authorityType": "enabled",
        "id": "refundDeposit",
        "name": "Cancel Permit",
      },
      Object {
        "authorityType": "enabled",
        "id": "viewEmailPrintPermit",
        "name": "Cancel Permit",
      },
      Object {
        "authorityType": "enabled",
        "id": "viewPrintAmendment",
        "name": "Cancel Permit",
      },
      Object {
        "authorityType": "disabled",
        "id": "editEvent",
        "name": "Edit Event",
      },
      Object {
        "authorityType": "disabled",
        "id": "addEvent",
        "name": "Add Event",
      },
      Object {
        "authorityType": "disabled",
        "id": "deleteEvent",
        "name": "Delete Event",
      },
      Object {
        "authorityType": "display",
        "id": "addCharge",
        "name": "Add Charge",
      },
      Object {
        "authorityType": "hide",
        "id": "deleteCharge",
        "name": "Delete Charge",
      },
      Object {
        "authorityType": "hide",
        "id": "editCharge",
        "name": "Edit Charge",
      },
      Object {
        "authorityType": "enabled",
        "id": "editQuestion",
        "name": "Edit Question",
      },
      Object {
        "authorityType": "enabled",
        "id": "editWaiver",
        "name": "Edit Waiver",
      },
      Object {
        "authorityType": "disabled",
        "id": "editNote",
        "name": "Edit Note",
      },
      Object {
        "authorityType": "disabled",
        "id": "addCustomerQuestions",
        "name": "Add Customer Questions",
      },
      Object {
        "authorityType": "disabled",
        "id": "addWaivers",
        "name": "Add Waivers",
      },
      Object {
        "authorityType": "enabled",
        "id": "resetFees",
        "name": "Reset Fees",
      },
    ]
  }
  errorHighlight={true}
  errors={Array []}
  eventActionInformation={
    Object {
      "addCharge": Object {
        "systemBehaviorID": "addCharge",
      },
      "addEvent": Object {
        "systemBehaviorID": "addEvent",
      },
      "addQuestions": Object {
        "systemBehaviorID": "addCustomerQuestions",
      },
      "addWaivers": Object {
        "systemBehaviorID": "addWaivers",
      },
      "deleteCharge": Object {
        "systemBehaviorID": "deleteCharge",
      },
      "deleteEvent": Object {
        "systemBehaviorID": "deleteEvent",
      },
      "deleteQuestion": Object {
        "systemBehaviorID": "deleteQuestion",
      },
      "editCharge": Object {
        "systemBehaviorID": "editCharge",
      },
      "editEvent": Object {
        "systemBehaviorID": "editEvent",
      },
      "editNote": Object {
        "systemBehaviorID": "editNote",
      },
      "editQuestion": Object {
        "systemBehaviorID": "editQuestion",
      },
      "editWaiver": Object {
        "systemBehaviorID": "editWaiver",
      },
      "resetFees": Object {
        "systemBehaviorID": "resetFees",
      },
    }
  }
  eventDetail={
    Immutable.Map {
      isShow: Immutable.Map {
        event_1: true,
      },
    }
  }
  eventIndex="1"
  facility={
    Immutable.Map {
      allFacilities: Immutable.Map {
        event_1: Immutable.Map {
          eventFee: Immutable.Map {
            facilityFees: Immutable.List [
            ],
          },
        },
      },
    }
  }
  hasRequiredQuestion={true}
  initialData={
    Object {
      "permitLabel": "",
    }
  }
  isPermitUpdatingMode={false}
  isTrrigerCustomerSearch={false}
  noteList={
    Immutable.Map {
      allNotes: Immutable.Map {
        event_1: Immutable.Map {
        },
      },
    }
  }
  questions={Array []}
  rules={Array []}
  waiverList={
    Immutable.Map {
      allWaivers: Immutable.Map {
        event_1: Immutable.Map {
        },
      },
      data: Immutable.List [
      ],
    }
  }
>
  <div
    className="event-detail reservation-fee content-error"
  >
    <div
      className="permit-fee"
    >
      <div
        className="reservation-fee"
      >
        <FeeSection
          disabledResetFee={false}
          eventIndex="1"
          facilityFees={Array []}
          feeActionStatus={
            Object {
              "allowAddFee": true,
              "allowDeleteFee": true,
              "allowEditFee": true,
            }
          }
          fetchPermitFee={[Function]}
        />
      </div>
    </div>
    <div
      className="event-detail__question leftMargin"
    >
      <EventSurvey
        addQuestion={[Function]}
        addQuestionList={Array []}
        canDelete={true}
        disabledAddQuestions={false}
        eventIndex="1"
        hasRequiredQuestion={true}
        hideCustomQuestionsSection={false}
        questions={Array []}
        readOnly={false}
      >
        <AUISurvey
          addQuestion={[Function]}
          addQuestionList={Array []}
          canDelete={true}
          changeQuestion={[Function]}
          deleteQuestion={[Function]}
          disabledAddQuestions={false}
          questions={Array []}
          readOnly={false}
          showQuestions={false}
        >
          <div />
        </AUISurvey>
      </EventSurvey>
    </div>
    <div
      className="event-detail__waiver leftMargin"
    >
      <Waiver
        disabledAddWaivers={false}
        eventDetail={
          Immutable.Map {
            isShow: Immutable.Map {
              event_1: true,
            },
          }
        }
        eventIndex="1"
        loadAddableWaivers={[Function]}
        permitWording=""
        readOnly={false}
        waiver={
          Immutable.Map {
            allWaivers: Immutable.Map {
              event_1: Immutable.Map {
              },
            },
            data: Immutable.List [
            ],
          }
        }
      >
        <div />
      </Waiver>
    </div>
    <div
      className="event-detail__notes leftMargin"
    >
      <Notes
        eventIndex="1"
        immediate={true}
        notes={
          Immutable.Map {
          }
        }
        readOnly={false}
      >
        <div
          className="notes-section u-hidden"
        >
          <Collapse
            activeKey=""
            dataSource={Array []}
            multiple={true}
            onChange={[Function]}
            prefixCls="an-collapse"
          >
            <div
              className="an-collapse"
            >
              <Panel
                Header="Event Notes"
                headerButtonId="default"
                isActive={false}
                key="notes"
                onItemClick={[Function]}
                prefixCls="an-collapse"
              >
                <div
                  className="an-collapse-item"
                >
                  <a
                    aria-expanded={false}
                    className="an-collapse-item__header"
                    href="javascript:void(0)"
                    onClick={[Function]}
                    role="button"
                    tabIndex={0}
                  >
                    <div
                      onClick={[Function]}
                    >
                      Event Notes
                      <i
                        className="icon icon-chevron-down"
                        onClick={[Function]}
                        tabIndex={0}
                      />
                    </div>
                  </a>
                  <div
                    className="an-collapse-item__content"
                  >
                    <div
                      className="section-container"
                    >
                      <section>
                        <label
                          htmlFor="staffNote"
                        >
                          Staff Note
                        </label>
                        <textarea
                          className=""
                          disabled={false}
                          id="staffNote"
                          maxLength="20000"
                          name="staffNote"
                          onBlur={[Function]}
                          onChange={[Function]}
                        />
                      </section>
                      <section>
                        <label
                          htmlFor="customerNote"
                        >
                          Customer Note
                        </label>
                        <textarea
                          className=""
                          disabled={false}
                          id="customerNote"
                          maxLength="20000"
                          name="customerNote"
                          onBlur={[Function]}
                          onChange={[Function]}
                        />
                      </section>
                    </div>
                  </div>
                </div>
              </Panel>
            </div>
          </Collapse>
        </div>
      </Notes>
    </div>
  </div>
</EventDetail>
`;

exports[`index -> ReservationDetail -> components -> EventDetail render component correctly if no data 1`] = `
<EventDetail
  addQuestionList={Array []}
  allEventConfig={
    Immutable.Map {
      hideReservationCharges: true,
      hideNotesSection: true,
      hideChecklistItemsSection: true,
      hideCustomQuestionsSection: true,
    }
  }
  authorities={
    Array [
      Object {
        "authorityType": "display",
        "id": "newReservationHome",
        "name": "New Reservation Home",
      },
      Object {
        "authorityType": "display",
        "id": "bookingGrid",
        "name": "Booking Grid",
      },
      Object {
        "authorityType": "enabled",
        "id": "permitActions",
        "name": "Permit Actions",
      },
      Object {
        "authorityType": "display",
        "id": "resourceCalendar",
        "name": "Resource Calendar",
      },
      Object {
        "authorityType": "enabled",
        "id": "bookingGridIcon",
        "name": "Booking Grid Icon",
      },
      Object {
        "authorityType": "enabled",
        "id": "reserveActions",
        "name": "Reserve Actions",
      },
      Object {
        "authorityType": "enabled",
        "id": "changePermitStatus",
        "name": "Change Permit Status",
      },
      Object {
        "authorityType": "disabled",
        "id": "cancelPermit",
        "name": "Cancel Permit",
      },
      Object {
        "authorityType": "disabled",
        "id": "modifyPaymentPlan",
        "name": "Cancel Permit",
      },
      Object {
        "authorityType": "enabled",
        "id": "makePayment",
        "name": "Cancel Permit",
      },
      Object {
        "authorityType": "disabled",
        "id": "refundCharges",
        "name": "Cancel Permit",
      },
      Object {
        "authorityType": "enabled",
        "id": "refundDeposit",
        "name": "Cancel Permit",
      },
      Object {
        "authorityType": "enabled",
        "id": "viewEmailPrintPermit",
        "name": "Cancel Permit",
      },
      Object {
        "authorityType": "enabled",
        "id": "viewPrintAmendment",
        "name": "Cancel Permit",
      },
      Object {
        "authorityType": "disabled",
        "id": "editEvent",
        "name": "Edit Event",
      },
      Object {
        "authorityType": "disabled",
        "id": "addEvent",
        "name": "Add Event",
      },
      Object {
        "authorityType": "disabled",
        "id": "deleteEvent",
        "name": "Delete Event",
      },
      Object {
        "authorityType": "display",
        "id": "addCharge",
        "name": "Add Charge",
      },
      Object {
        "authorityType": "hide",
        "id": "deleteCharge",
        "name": "Delete Charge",
      },
      Object {
        "authorityType": "hide",
        "id": "editCharge",
        "name": "Edit Charge",
      },
      Object {
        "authorityType": "enabled",
        "id": "editQuestion",
        "name": "Edit Question",
      },
      Object {
        "authorityType": "enabled",
        "id": "editWaiver",
        "name": "Edit Waiver",
      },
      Object {
        "authorityType": "disabled",
        "id": "editNote",
        "name": "Edit Note",
      },
      Object {
        "authorityType": "disabled",
        "id": "addCustomerQuestions",
        "name": "Add Customer Questions",
      },
      Object {
        "authorityType": "disabled",
        "id": "addWaivers",
        "name": "Add Waivers",
      },
      Object {
        "authorityType": "enabled",
        "id": "resetFees",
        "name": "Reset Fees",
      },
    ]
  }
  errorHighlight={false}
  errors={Array []}
  eventActionInformation={
    Object {
      "addCharge": Object {
        "systemBehaviorID": "addCharge",
      },
      "addEvent": Object {
        "systemBehaviorID": "addEvent",
      },
      "addQuestions": Object {
        "systemBehaviorID": "addCustomerQuestions",
      },
      "addWaivers": Object {
        "systemBehaviorID": "addWaivers",
      },
      "deleteCharge": Object {
        "systemBehaviorID": "deleteCharge",
      },
      "deleteEvent": Object {
        "systemBehaviorID": "deleteEvent",
      },
      "deleteQuestion": Object {
        "systemBehaviorID": "deleteQuestion",
      },
      "editCharge": Object {
        "systemBehaviorID": "editCharge",
      },
      "editEvent": Object {
        "systemBehaviorID": "editEvent",
      },
      "editNote": Object {
        "systemBehaviorID": "editNote",
      },
      "editQuestion": Object {
        "systemBehaviorID": "editQuestion",
      },
      "editWaiver": Object {
        "systemBehaviorID": "editWaiver",
      },
      "resetFees": Object {
        "systemBehaviorID": "resetFees",
      },
    }
  }
  eventDetail={
    Immutable.Map {
      isShow: Immutable.Map {
        event_1: true,
      },
    }
  }
  eventIndex="2"
  facility={
    Immutable.Map {
      allFacilities: Immutable.Map {
        event_1: Immutable.Map {
          eventFee: Immutable.Map {
            facilityFees: Immutable.List [
            ],
          },
        },
      },
    }
  }
  hasRequiredQuestion={true}
  initialData={
    Object {
      "permitLabel": "",
    }
  }
  isPermitUpdatingMode={false}
  isTrrigerCustomerSearch={false}
  noteList={
    Immutable.Map {
      allNotes: Immutable.Map {
        event_1: Immutable.Map {
        },
      },
    }
  }
  questions={Array []}
  rules={Array []}
  waiverList={
    Immutable.Map {
      allWaivers: Immutable.Map {
        event_1: Immutable.Map {
        },
      },
      data: Immutable.List [
      ],
    }
  }
>
  <div
    className="event-detail u-hidden "
  >
    <div
      className="event-detail__question leftMargin"
    >
      <EventSurvey
        addQuestion={[Function]}
        addQuestionList={Array []}
        canDelete={true}
        disabledAddQuestions={false}
        eventIndex="2"
        hasRequiredQuestion={true}
        hideCustomQuestionsSection={true}
        questions={Array []}
        readOnly={false}
      >
        <AUISurvey
          addQuestion={[Function]}
          addQuestionList={Array []}
          canDelete={true}
          changeQuestion={[Function]}
          deleteQuestion={[Function]}
          disabledAddQuestions={false}
          questions={Array []}
          readOnly={false}
          showQuestions={false}
        >
          <div />
        </AUISurvey>
      </EventSurvey>
    </div>
  </div>
</EventDetail>
`;
